### PR TITLE
Update to Django 4.2

### LINKED ISSUE #749: Update to Django 4.2 LTS
A new LTS version has been released for Django!

Our current version (3.2) will be in extended security support until April 2024.

https://www.djangoproject.com/weblog/2023/apr/03/django-42-released/

### PR DESCRIPTION
**Pull request by bramvandartel**
Please add the following information to your pull request:

**Please describe what your PR is fixing**
Update to Django 4.2

**Concretely, which issues does your PR solve? (Please reference them by typing `Fixes/References Inter-Actief/amelie#<issue_id>`)**
Fixes #749 

**Does your PR change how we process personal data, impact our [privacy document](https://www.inter-actief.utwente.nl/privacy/), or modify (one of) our data export(s)?**
no

**Does your PR include any django migrations?**
yes (and some that were still open on main)

**Does your PR include the proper translations (did you add translations for new/modified strings)?**
no, my PR does not include translations

**Does your PR include CSS changes (and did you run the `compile_css.sh` script in the `scripts` directory to regenerate the `compiled.css` file)?**
no, my PR does not include CSS changes

**Does your PR need external actions by for example the System Administrators? (Think about new pip packages, new (local) settings, a new regular task or cronjob, new management commands, etc.)?**
yes, update of python version, pip upgrades

**Did you properly test your PR before submitting it?**
no

### CODE REVIEW

**Comment by bramvandartel** [comment]
I noticed @supertom01 has been working on a Django 4.2 upgrade as well. The issue, #749, has been open for a while and support for Django 3.2 has been dropped as of 1st of April.

What still has to be done is:

- [ ] Extensive testing of all webpages (the manage.py test runs successfully)
- [ ] Look at all the pip packages in the requirements and see which can be updated. This might be a good moment to also fix dependabot alerts.
- [ ] Beheer should perform upgrades on the servers

**Comment by supertom01** [comment]
I've checked the pip packages that can be updated, however these changes should still be thoroughly tested. I haven't had any time to setup a local testing environment yet. So this will have to happen later

File: amelie/urls.py [hunk 1]
```diff
@@ -137,11 +135,13 @@
 
 
 if settings.DEBUG_TOOLBAR:
-    import debug_toolbar
+    # import debug_toolbar
```
Comment (Kurocon) [inline amelie/urls.py:138]: Commented out code

File: amelie/urls.py [hunk 2]
```diff
@@ -137,11 +135,13 @@
...
+    from debug_toolbar.toolbar import debug_toolbar_urls
     # Django debug toolbar
-    urlpatterns += [
-        path('__debugtoolbar__/', include(debug_toolbar.urls), name='django_debug_toolbar')
-    ]
+    urlpatterns += debug_toolbar_urls()
+    # [
+    #     path('__debugtoolbar__/', include(debug_toolbar.urls), name='django_debug_toolbar')
```
Comment (Kurocon) [inline amelie/urls.py:143]: Commented out code

File: requirements.txt [hunk 3]
```diff
@@ -1,15 +1,14 @@
-Django>=3.2.16,<3.3
-
+Django>=4.2.11,<4.3
+# Django>=3.2,<3.3
```
Comment (Kurocon) [inline requirements.txt:2]: Commented out code

File: requirements.txt [hunk 4]
```diff
@@ -21,36 +20,40 @@ django-environ
...
-django-ckeditor>=6.3.2,<6.4  # 6.4+ requires Python3.8+ and Django 3.2+
+django-ckeditor>=6.3.2
 
 # FCM (Firebase Cloud Messaging)
-fcm-django>=0.3.11,<0.4  # 1.x has breaking changes (old firebase api removed)
+fcm-django>=0.3.11,<1  # 1.x has breaking changes (old firebase api removed) # TODO: Breaking
 
 # Custom forks or specific commits of django packages
-git+https://github.com/Inter-Actief/django-sslserver@204586ce686212a4c2848c12f88f7279802b5dcc
+# git+https://github.com/Inter-Actief/django-sslserver@204586ce686212a4c2848c12f88f7279802b5dcc
```
Comment (Kurocon) [inline requirements.txt:29]: Commented out code

**Review by Kurocon — APPROVED** [review]

**Pull request merged**